### PR TITLE
refactor: rename fake package

### PR DESCRIPTION
**Pull request by sdowell**
This package is not only used in testing, so the path/naming is a bit misleading. This package primarily contains helper functions to construct k8s objects.

### CODE REVIEW

File: pkg/core/k8sobjects/cluster_config.go
Comment (nan-yu) [inline pkg/core/k8sobjects/cluster_config.go:1]: ClusterConfig is a CRD used by the monorepo only. We can get rid of it in a separate PR.

File: pkg/core/k8sobjects/errors.go
Comment (nan-yu) [inline pkg/core/k8sobjects/errors.go:1]: This should go outside of the `k8sobjects` package.

File: pkg/core/k8sobjects/namespace_config.go
Comment (nan-yu) [inline pkg/core/k8sobjects/namespace_config.go:1]: NamespaceConfig is also monorepo only, and can be removed in a separate PR.

File: pkg/core/k8sobjects/repo.go
Comment (nan-yu) [inline pkg/core/k8sobjects/repo.go:1]: Another monorepo only CR.

File: pkg/core/k8sobjects/errors.go
Comment (sdowell) [reply pkg/core/k8sobjects/errors.go:1]: Done

File: pkg/core/k8sobjects/cluster_config.go
Comment (sdowell) [reply pkg/core/k8sobjects/cluster_config.go:1]: Ack

File: pkg/core/k8sobjects/namespace_config.go
Comment (sdowell) [reply pkg/core/k8sobjects/namespace_config.go:1]: Ack

File: pkg/core/k8sobjects/repo.go
Comment (sdowell) [reply pkg/core/k8sobjects/repo.go:1]: Ack

**Review by nan-yu — APPROVED** [review]
/lgtm
